### PR TITLE
feat: vault bulk en/decrypt

### PR DESCRIPTION
**Pull request by Flo4604**
## What does this PR do?

Adds bulk encryption and decryption endpoints to the Vault service to improve performance when processing multiple items. The new endpoints allow encrypting/decrypting multiple values in a single RPC call instead of making individual calls for each item.

The changes include:
- New `EncryptBulk` and `DecryptBulk` RPC methods in the Vault service
- Protobuf definitions for bulk request/response messages with map-based item handling
- Updated API routes to use bulk decryption for listing keys with decrypt option
- Updated Krane secrets service to use bulk decryption for environment variables

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Enhancement (small improvements)

## How should this be tested?

- Test bulk encryption by calling the new `EncryptBulk` endpoint with multiple items and verify all are encrypted correctly
- Test bulk decryption by calling the new `DecryptBulk` endpoint with multiple encrypted values and verify all are decrypted correctly
- Test roundtrip encryption/decryption using bulk endpoints to ensure data integrity
- Test API key listing with decrypt option to verify bulk decryption integration
- Test Krane secrets decryption to verify bulk decryption integration
- Verify authentication and authorization still work correctly for bulk endpoints
- Test with empty item maps to ensure proper handling of edge cases

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Ran `make fmt` on `/go` directory
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] Updated the Unkey Docs if changes were necessary

### CODE REVIEW

**Comment by Flo4604** [comment]
* **#5127** <a href="https://app.graphite.com/github/pr/unkeyed/unkey/5127?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/unkeyed/unkey/5127?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: svc/api/routes/v2_apis_list_keys/handler.go [hunk 1]
```diff
@@ -235,20 +235,25 @@ func (h *Handler) Handle(ctx context.Context, s *zen.Session) error {
...
+				Keyring: auth.AuthorizedWorkspaceID,
+				Items:   bulkItems,
+			})
+			if bulkErr != nil {
+				logger.Error("failed to bulk decrypt keys",
+					"error", bulkErr,
+				)
+			} else {
+				for id, plaintext := range bulkRes.GetItems() {
+					plaintextMap[id] = plaintext
```
Comment (chronark) [inline svc/api/routes/v2_apis_list_keys/handler.go]: what'st he point in looping here and assigning this to plaintextMap, instead we could juse use bulkRes directly further down
